sso/permissions.py

```diff
@@ -1,29 +1,32 @@
 from unicoresso.models import AuthorizedSite
+from urlparse import urlparse
+import re
 
 
 def custom_attributes(user, service):
-    # for group in request.user.groups.all():
-    #     if (service in group.objects.all()):
-    # return {'givenName': user.first_name, 'email': user.email, 'has_perm':
-    # 'yes'}
-    groups = user.groups.values_list('name', flat=True)
+    service_domain = urlparse(service).netloc
 
-    if AuthorizedSite.objects.filter(site=service).exists():
-        website = AuthorizedSite.objects.get(site=service)
-        site_groups = website.group.values_list('name', flat=True)
+    matched_sites = [
+        s for s in AuthorizedSite.objects.all()
+        if re.match(
+            re.compile(s.site.replace('.', '\\.').replace('*', '.*')),
```

Comment:
Also:

```python
>>> from glob.fnmatch import fnmatch
>>> fnmatch.fnmatch('http://bar.foo.com', 'http://*.foo.com')
True
>>>
```

Manually escaping regex in any code that relates to auth makes me quite nervous.